Fix Dashboard hover-over action so it shows the correct info

kialam opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Related #2533

This PR fixes the on-hover tooltip anchoring to only the first point on the Dashboard. Our newest version fo Node 8 requires upgrades to our D3 and NVD3 libraries. NVD3 v1.8.1 introduces [refactored tooltips](https://github.com/novus/nvd3/commit/3d9f5d8821f5617d6ec72f30cf04c981d90bdc97). As part of this PR, the job chart dashboard directive has been refactored to account for the data changes that the NVD3 library expects.
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - UI

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 2.1.0
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
![dashboard_hover](https://user-images.githubusercontent.com/2293210/48149170-753b0800-e289-11e8-8adb-7f1b8da0cb67.gif)

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

mabashian commented:
Code changes look good to me.  Is it normal that the package-lock.json file would change this much from pinning those two dependencies?  If that's expected 👍

kialam committed:
Fix Dashboard hover-over action so it shows the correct info.

- Refactor our Dashboard directive to display the dashboard graph's x-axis according to the new changes in NVD3 v1.8.1. See https://github.com/nvd3-community/nvd3/blob/gh-pages/examples/lineChart.html for reference.

kialam commented:
@mabashian it looks like the package-lock.json file is supposed to update any time `npm install` is run. ~~I think because our `make ui-docker` script runs `npm --unsafe-perm --prefix awx/ui install --no-save awx/ui` which kicks off the package-lock.json being updated if any dependencies have been upgraded.~~ In the process of working on this I did pin our dependency version of D3 and NVD3 using `npm install` e.g.: `npm install --save --save-exact d3@3.5.4`.

The good news is the refactoring of the job chart dashboard directive works on both the most recent versions of D3 and NVD3 (which is what we are currently running) so I no longer see a need to pin those dependencies down like before.

jlmitch5 reviewed (approved):
very nice!

jakemcdermott commented:
recheck
